### ISSUE TITLE
TypeError: publicProperty.getType is not a function

### ISSUE THREAD

**Issue by vitalikmay** [FroggyPanda/better-supabase-types#33]
Hello,

Package: `better-supabase-types@2.7.2`
Error:
```
/project/node_modules/better-supabase-types/dist/index.js:128
  const publicType = publicProperty.getType();
                                    ^

TypeError: publicProperty.getType is not a function
    at getTablesProperties (/project/node_modules/better-supabase-types/dist/index.js:128:37)
    at generate (/project/node_modules/better-supabase-types/dist/index.js:240:30)
    at Object.handler (/project/node_modules/better-supabase-types/dist/index.js:444:7)
    at /project/node_modules/yargs/build/index.cjs:1:8993
    at j (/project/node_modules/yargs/build/index.cjs:1:4956)
    at _.handleValidationAndGetResult (/project/node_modules/yargs/build/index.cjs:1:8962)
    at _.applyMiddlewareAndGetResult (/project/node_modules/yargs/build/index.cjs:1:9604)
    at _.runCommand (/project/node_modules/yargs/build/index.cjs:1:7231)
    at [runYargsParserAndExecuteCommands] (/project/node_modules/yargs/build/index.cjs:1:58912)
    at te.parse (/project/node_modules/yargs/build/index.cjs:1:40478)
```

The types file [types.txt](https://github.com/FroggyPanda/better-supabase-types/files/14293399/types.txt)

**Comment by PixsaOJ** [comment]
I have tried to run  against your `types.txt` but it only gives me
`warn No enums found within the Enums property for schema public.`

Could you specify more details such as node version and package manager.

I am checking on commit ef1e5f0 this is before changes were reverted

**Comment by PixsaOJ** [comment]
Only way that `getType` does not exists, is if `getPropertyOrThrow` method from **_ts-morph_** would return something else than `PropertySignature`. There are no IDE errors, so return types for  `getPropertyOrThrow(name: string)` must be wrong somehow.

I tried renaming input file to `*.txt` but it gave different errors. Can't seem to reproduce this error.

**Comment by FroggyPanda** [comment]
I am so sorry; I thought I had commented on this issue.  But same as PixasOJ, I also had trouble reproducing this issue but still went ahead to revert the changes.

**Comment by PixsaOJ** [comment]
In current state, library is broken because its looking for interface. 

If i can not reproduce this, it means that some people can use the library. I think move would be to publish the changes, this way at least some people can use it.

Currently hack is to manually rename Supabase generated type to interface.

**Comment by FroggyPanda** [comment]
Yeah, I was thinking I was doing something wrong.  But if you and I cannot get the same results, I will wait for a response before I push out the PR again and see if we can solve this issue.  If not, I feel like people won't have a problem switching `type` to `interface` as the structure of the database object has not seemed to change.

**Comment by vitalikmay** [comment]
I use `npm` and `Node.js v18.14.0`. By the way, I renamed `types.ts` to `types.txt` because GitHub does not allow attaching `*.ts` files.

I wanted to prepare a test project to reproduce the issue, but I found out that you had already removed `better-supabase-types@2.7.2`. 

So, I tried using `better-supabase-types@2.7.1` and encountered the error: `InvalidOperationError: Expected to find interface named 'Database'`. As @FroggyPanda mentioned, manually switching the `Database` from `type` to `interface` fixed the error.

**Comment by FroggyPanda** [comment]
> I use `npm` and `Node.js v18.14.0`. By the way, I renamed `types.ts` to `types.txt` because GitHub does not allow attaching `*.ts` files.
> 
> I wanted to prepare a test project to reproduce the issue, but I found out that you had already removed `better-supabase-types@2.7.2`.
> 
> So, I tried using `better-supabase-types@2.7.1` and encountered the error: `InvalidOperationError: Expected to find interface named 'Database'`. As @.FroggyPanda mentioned, manually switching the `Database` from `type` to `interface` fixed the error.

If you want to, try downloading the PR (#32) and see if you're able to reproduce the issue with that.

**Comment by PixsaOJ** [comment]
@FroggyPanda OP not responding, maybe its worth publishing?

**Comment by FroggyPanda** [comment]
> @.FroggyPanda OP not responding, maybe its worth publishing?
@PixsaOJ, that is what I was thinking.  Been busy with school so it slipped my mind how long its been since OP has responded.

**Comment by FroggyPanda** [comment]
Version [2.7.3](https://www.npmjs.com/package/better-supabase-types/v/2.7.3) has been published (had to bump the version from 2.7.2 to 2.7.3 because you cannot republish the same version number that was unpublished).

**Comment by PixsaOJ** [comment]
Works like a charm 🍓
